src/Symfony/Component/Validator/Constraint.php

```diff
@@ -110,6 +116,28 @@ public function __construct(mixed $options = null, ?array $groups = null, mixed
     {
         unset($this->groups); // enable lazy initialization
 
+        self::$checkedConstraintClasses[static::class] ??= str_starts_with(static::class, 'Symfony\Component\Validator\Constraints\\') || (new \ReflectionClass(static::class))->getAttributes(InitializesOptions::class) !== [];
```

Comment:
I don't think we should have this check here. We should deprecate entirely passing non-null options.